src/display/annotation_layer.js

```diff
@@ -858,14 +846,37 @@ class TextWidgetAnnotationElement extends WidgetAnnotationElement {
    *
    * @private
    * @param {HTMLDivElement} element
-   * @param {Object} font
    * @memberof TextWidgetAnnotationElement
    */
-  _setTextStyle(element, font) {
-    // TODO: This duplicates some of the logic in CanvasGraphics.setFont().
+  _setTextStyle(element) {
```

Comment:
This PR should have a reference test attached to it so we can avoid future regressions. One of the PDF files from the issues you mentioned will suffice, or another file you may have that is smaller and shows the same issue that this PR fixes. Since this shows up when rendering forms, you can add a test similar to https://github.com/mozilla/pdf.js/commit/adf0972ca57e66202acb489b8782a3b73d510bf5#diff-bbeef90df04bff7617f7ef59e3f37cac92192c5169bc500d857bd6fb107fcabd: add the PDF file, add a gitignore entry and add a manifest entry with `"forms": true` to enable form rendering for your reference test.